osm2gtfs/core/elements.py

```diff
@@ -0,0 +1,186 @@
+    a Line object.
+
+    In OpenStreetMap this is usually represented as "route" relation.
+    In GTFS this is not exlicitly presented but used as base to create "trips"
+
+    """
+    route_id = attr.ib()
+    shape = attr.ib()
+
+    line = attr.ib(default=None)
+    fr = attr.ib(default=None)
+    to = attr.ib(default=None)
+    duration = attr.ib(default=None)
+
+    # All stop objects of itinerary
+    stops = attr.ib(default=attr.Factory(list))
+
+    def __attrs_post_init__(self):
+        '''
+        Populates the object with information obtained from the tags
+        '''
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'from' in self.tags:
+            self.fr = self.tags['from']
+
+        # pylint: disable=unsupported-membership-test,unsubscriptable-object
+        if 'to' in self.tags:
+            self.to = self.tags['to']
+
+    def get_stops(self):
+        return self.stops
+
+
+@attr.s
+class Station(Element):
+    """A public transport stop of the type station.
+
+    It's a representation of a possible group of stops.
+
+    In OpenStreetMap this is usually represented as "stop_area" relation.
+    In GTFS it is handled as a stop with a location_type=1. Regular Stops with
+    location_type=0 might specify a station as parent_station.
+
+    """
+    lat = attr.ib()
+    lon = attr.ib()
+
+    stop_id = attr.ib(default="")
+    location_type = attr.ib(default=1)
+
+    # Stops forming part of this Station
+    _members = attr.ib(default=attr.Factory(list))
+
+    def set_members(self, members):
+        self._members = members
+
+    def get_members(self):
+        return self._members
+
+    def get_stop_id(self):
+        return self.stop_id
+
+    def set_stop_id(self, stop_id):
+        self.stop_id = stop_id
+
+
+@attr.s
+class Stop(Element):
+    """A public transport stop.
+
+    In OpenStreetMap this is usually represented as an object of the role
+    "plattform" in the route.
+
+    """
+    lat = attr.ib()
+    lon = attr.ib()
+
+    stop_id = attr.ib("")
+    location_type = attr.ib(default=0)
+
+    # The id of the Station this Stop might be part of.
+    _parent_station = attr.ib(default=None)
+
+    def set_parent_station(self, identifier, override=False):
+        """
+        Set the parent_station_id on the first time;
+        Second attempts throw a warning
+        """
+        if self._parent_station is None or override is True:
+            self._parent_station = identifier
+        else:
+            sys.stderr.write("Warning: Stop is part of two stop areas:\n")
+            sys.stderr.write(
+                "https://osm.org/" + self.osm_type + "/" + str(
+                    self.osm_id) + "\n")
+
+    def get_parent_station(self):
+        return self._parent_station
+
+    def get_stop_id(self):
+        return self.stop_id
+
+    def set_stop_id(self, stop_id):
+        self.stop_id = stop_id
```

Comment:
Oh, yeah, big question on having `elements` or `routes` and `stops`. I also liked routes and stops, until I got the request to abstract the common attributes and I moved them into `Element`. Then I though, oh three files: hmmm - no, two files with one in there - hmmmm, inconsisten, and then I put them into one. I'm not strong with this one. What would you suggest?

Reply:
Couldn't you just have three files: elements (or osm_base), osm_routes and osm_stops?

Reply:
I guess this is because of individual taste... I preferred to put them all in one file. But I'm not that strong about this. So, renamed them to `elements.py`, `routes.py`, `stops.py`.

By the way, it doesn't make sense to use the prefix `osm_` because the data structure on purpose was made with the motivation to abstract it from OSM particular things instead of having the long spaghetti. All this is living now nicely separated into `osm_connector.py`.

Reply:
Did you see this:

> The reason why the modules should not be just names routes and stops, is because then variable names in the code shadow module names.

So in Python it is then ambiguous if `routes` means the variable or the `routes` module. Therefore, it is better to give a different name to the module (or rename all `stops` and `routes` variables used throughout the code).

Reply:
So, why not staying with one `elements.py`?